frontend/src/lib/components/BillingAlerts.tsx

```diff
@@ -1,60 +1,83 @@
+        message = (
+            <p>
+                <b>Warning!</b> You have already used{' '}
+                <b style={{ color: 'var(--warning)' }}>{freePlanPercentage && freePlanPercentage * 100}%</b> of your 1
+                million free events this month.{' '}
+                <Link to="/organization/billing" data-attr="alert_free_usage_near_limit">
+                    {billing?.plan?.custom_setup_billing_message ||
+                        'To avoid losing data or access to it, upgrade your billing plan now.'}
+                </Link>
+            </p>
+        )
+    }
 
-                    <LinkButton
-                        type="primary"
-                        to={
-                            alertToShow === BillingAlertType.SetupBilling
-                                ? billing?.subscription_url
-                                : '/organization/billing'
-                        }
-                    >
-                        <ToolFilled /> {alertToShow === BillingAlertType.SetupBilling ? 'Set up now' : 'Manage billing'}
-                    </LinkButton>
-                </div>
-            </div>
-        </>
+    if (alertToShow === BillingAlertType.SetupBilling) {
+        isWarning = true
+        message = (
+            <p>
+                <b>Action needed!&nbsp;</b>
+                <Link href={billing?.subscription_url} data-attr="alert_setup_billing">
+                    {billing?.plan?.custom_setup_billing_message ||
+                        'Please finish setting up your billing information.'}
+                </Link>
+            </p>
+        )
+    }
+
+    if (alertToShow === BillingAlertType.UsageNearLimit) {
+        isWarning = true
+        message = (
+            <p>
+                <b>Warning!</b> You have already used{' '}
+                <b style={{ color: typeof strokeColor === 'string' ? strokeColor : 'inherit' }}>
+                    {percentage && percentage * 100}%
+                </b>{' '}
+                of your event allocation this month.{' '}
+                <Link to="/organization/billing" data-attr="alert_usage_near_limit">
+                    {billing?.plan?.custom_setup_billing_message ||
+                        'To avoid losing data or access to it, upgrade your billing plan now.'}
+                </Link>
+            </p>
+        )
+    }
+
+    if (alertToShow === BillingAlertType.UsageLimitExceeded) {
+        isAlert = true
+        message = (
+            <p>
+                <b>Alert!</b> The monthly limit of events or billing limit for your organization has been exceeded.{' '}
+                <Link to="/organization/billing" data-attr="alert_usage_limit_exceeded">
+                    {billing?.plan?.custom_setup_billing_message ||
+                        'To avoid losing data or access to it, increase your billing limit now.'}
+                </Link>
+            </p>
+        )
+    }
+
+    return (
+        <div className={'Announcement'}>
```

Comment:
Not a fan of using a scss class from a different component tbh. Makes it hard to know what effect changes are having on other places.